omni-relayer/src/workers/solana.rs

```diff
@@ -0,0 +1,290 @@
+    let redis_connection = redis_client.get_multiplexed_tokio_connection().await?;
+
+    loop {
+        let mut redis_connection = redis_connection.clone();
+
+        let events = match utils::redis::get_events(
+            &mut redis_connection,
+            utils::redis::SOLANA_INIT_TRANSFER_EVENTS.to_string(),
+        )
+        .await
+        {
+            Some(events) => events,
+            None => {
+                tokio::time::sleep(tokio::time::Duration::from_secs(
+                    utils::redis::SLEEP_TIME_AFTER_EVENTS_PROCESS_SECS,
+                ))
+                .await;
+                continue;
+            }
+        };
+
+        let mut handlers = Vec::new();
+
+        for (key, event) in events {
+            if let Ok(init_transfer_with_timestamp) =
+                serde_json::from_str::<InitTransferWithTimestamp>(&event)
+            {
+                let handler = handle_init_transfer_event(
+                    config.clone(),
+                    connector.clone(),
+                    redis_connection.clone(),
+                    key.clone(),
+                    init_transfer_with_timestamp,
+                );
+                handlers.push(tokio::spawn(handler));
+            }
+        }
+
+        join_all(handlers).await;
+
+        tokio::time::sleep(tokio::time::Duration::from_secs(
+            utils::redis::SLEEP_TIME_AFTER_EVENTS_PROCESS_SECS,
+        ))
+        .await;
+    }
+}
+
+async fn handle_init_transfer_event(
+    config: config::Config,
+    connector: Arc<OmniConnector>,
+    mut redis_connection: redis::aio::MultiplexedConnection,
+    key: String,
+    init_transfer_with_timestamp: InitTransferWithTimestamp,
+) {
+    let current_timestamp = chrono::Utc::now().timestamp();
+
+    if current_timestamp
+        - init_transfer_with_timestamp
+            .last_update_timestamp
+            .unwrap_or_default()
+        < utils::redis::CHECK_INSUFFICIENT_FEE_TRANSFERS_EVERY_SECS
+    {
+        return;
+    }
+
+    if current_timestamp - init_transfer_with_timestamp.creation_timestamp
+        > utils::redis::KEEP_INSUFFICIENT_FEE_TRANSFERS_FOR
+    {
+        warn!(
+            "Removing an old InitTransfer on Solana: {:?}",
+            init_transfer_with_timestamp
+        );
+        utils::redis::remove_event(
+            &mut redis_connection,
+            utils::redis::SOLANA_INIT_TRANSFER_EVENTS,
+            &key,
+        )
+        .await;
+        return;
+    }
+
+    info!("Trying to process InitTransfer log on Solana");
+
+    let recipient = match init_transfer_with_timestamp
+        .recipient
+        .parse::<OmniAddress>()
+    {
+        Ok(recipient) => recipient,
+        Err(_) => {
+            warn!(
+                "Failed to parse recipient as OmniAddress: {:?}",
+                init_transfer_with_timestamp.recipient
+            );
+            return;
+        }
+    };
+
+    // TODO: Use existing API to check if fee is sufficient here
```

Comment:
Am I correct in understanding that we currently don’t check whether the fee is acceptable for Solana and finalize all transactions?

Reply:
For Solana and Evm and Near. We don't check fee at all. As I said during the call, it's easy to add, but it'll slow down the debug process, so I decided to add it after all methods will be tested

Reply:
Better to add it anyway. You could, for example, add a feature like "skip_fee_check".

Reply:
Yep, but let's make another PR for that

Reply:
https://github.com/Near-One/omni-bridge/issues/162